tests/benchmarks/test_array.py

```diff
@@ -257,3 +258,29 @@ def test_map_overlap_sample(small_client, new_array):
     x = new_array((10000, 10000), chunks=(50, 50))  # 40_000 19.5 kiB chunks
     y = x.map_overlap(lambda x: x, depth=1)
     y[5000:5010, 5000:5010].compute()
+
+
+def _create_indexer(n, chunk_n):
+    idx = np.arange(0, n)
+    np.random.shuffle(idx[: n // 10])
+
+    indexer = []
+    for i in range(0, n, chunk_n):
+        indexer.append(idx[i : i + chunk_n].tolist())
+    return indexer
+
+
+def test_take(small_client, new_array):
```

Comment:
Is this test based on some real-world use case? If so, some context would be helpful.

Reply:
this is the smallest thing that the current implementation can still handle in reasonable time... everything bigger blows up